### ISSUE TITLE
Incorporate fixes for 1.3.2 into the new release

### ISSUE THREAD

**Issue by stoecker** [trusteddomainproject/OpenDMARC#159]
For our Suse release I'm packing OpenDMARC with these patches for a long time now as opendmarc development was stalled and the patches are more or less essential to get a working version.

http://batleth.sapienti-sat.org/projects/opendmarc/

It seems opendmarc is developed again, so please incorporate the patch set into the release version.

**Comment by thegushi** [comment]
These are not going to make it into 1.4.1.  We are literally days away from release of that and many of these need to be QA’d against various releases.

The most major of the fixes are already present in 1.4.0 or 1.4.1 now.

However, most of these probably will be accepted in the next few weeks either as a 1.4.2 or a 1.5.0 (which is where we’d expect new features to show up).

-Dan

**Comment by andreasschulze** [comment]
I support that request, too. Years ago I started with 1.3.1 and ended up with a similar page:
https://andreasschulze.de/dmarc/opendmarc

**Comment by martinbogo** [comment]
@thegushi has set a goal for things after 1.4.1(.1) ... and as he is the
patchmaster, He Has Spoken(tm)

We're going to concentrate on OpenDKIM for this cycle, then come back to
OpenDMARC and start testing and merging patches into 1.5.0.   OpenDKIM also
needs some long-overdue care and tending.

The same process we used for 1.4.x will apply going forward:

Individual issues will be patched and tracked, and tagged, with the
next release.   Some will be targeted to 1.4.2, some to 1.5.0 depending on
if they are bugfixes or enhancements, etc.  Again that's @thegushi's call
as to where they will slot in.  Yes, it means more work to do them one by
one that way -- but since we've made some changes into 1.4.1(.1) that means
that each one needs retesting.  There were a LOT of fixes put into this
last release.

**Comment by stoecker** [comment]
Ok. A year later and nothing done. Will it be the same situation as before that opendmarc simply is unmaintained and the number of patches piles up? Does it even make sense to report bugs?

When the current maintainers don't have time, so please choose somebody who can help you, but don't let the software die because you can't maintain it!

**Comment by glts** [comment]
@stoecker I also follow this project and agree completely.

Maintainers: There are many benign and well-motivated pull requests
open, some of them have been applied in distros like Debian and others.
It really would be high time to get these into mainline. If you don’t
have time, why not let someone else do it?

----

A comment on this issue here: Just linking to a page with patches for an
older release does not seem that helpful. Many such patch collections
exist for distros like Debian, Fedora, etc. The work here would be
assessing the patches and bringing them into shape for merging into the
current `develop` branch. (Otherwise you risk sounding a bit like ‘here
is a to-do list, now do it’.)

**Comment by stoecker** [comment]
> work here would be assessing the patches and bringing them into shape

Well, that's clear. I'd even would help doing so for my own patches and maybe others, but only when the effort is not wasted as it looks like now.

**Comment by stoecker** [comment]
Not a single commit since December and there are tons of existing patches and problems. Is the GitHub project as dead as the Sourceforge project was?

Will the next step be a release of 1.5.0 on GitLab without fixing any existing issue and restarting the bug tracker again?

**Comment by thegushi** [comment]
No, but it's volunteer managed and my time management (and life) is complicated.  The insults don't help, really.

-Dan

**Comment by stoecker** [comment]
As already said before: When you have no time yourself choose one of the many people who did work for the project already and let them help you.

And feeling insulted by stating of facts doesn't help either really.

**Comment by stoecker** [comment]
Time flies. 8 months gone again and nothing happened. Please let others help if you have no time!

**Comment by thegushi** [comment]
To everyone who waited and expressed frustration here: you were right to be frustrated, and I'm sorry it took this long.

Since May 2026, the develop branch has seen significant activity - over 30 pull requests merged covering security fixes, RFC compliance gaps, crash and memory-safety bugs, aggregate reporting improvements, and build system fixes. A summary is being maintained at https://github.com/thegushi/OpenDMARC/blob/develop/CHANGES-develop.md and a release is being prepared.

@stoecker - your record-grouping fix (messages with identical source IP and policy outcome should collapse into one report record with an accurate count rather than N records with count=1) landed this week as part of this effort. Thank you for the patience.

There is also a concrete external forcing function now: RFC 9989 obsoletes RFC 7489 and RFC 9990 updates the aggregate reporting spec. DMARCbis compliance isn't an excuse for the past delays, but it does mean a release has to happen - the protocol is moving and deployments need an updated implementation.

If you have patches - whether from your own downstream packaging, distro patch sets, or new work - please submit them as pull requests against the `develop` branch. I will review and merge them. The backlog that made this project look dead is mostly cleared; new contributions won't sit for years.

**Comment by thegushi** [comment]
Closing this one - the original report was a link to a third-party patch collection page, and the right path forward is individual PRs against develop so each fix can be reviewed, tested, and tracked on its own merits. If any of those patches haven't made it in yet, please open a PR and it will get attention. The door is open.

**Comment by stoecker** [comment]
It would really be easier if you simply use the patches and apply them yourself. It seems none of them have been added and as you can see they fix tickets of the old sourceforge ticket system.

http://batleth.sapienti-sat.org/projects/opendmarc/

**Comment by stoecker** [comment]
openSUSE uses ticket159.patch, ticket182.patch, ticket183.patch, ticket184.patch, ticket193.patch, ticket207.patch, ticket212.patch, z01_changeSubjectFailureReport_v2.patch, z03_reportDestVerificationV2.patch
in addition to the above PRs.

**Comment by thegushi** [comment]
Except that some of the sourceforge tickets were filed by different people, sourceforge is harder to parse, and we've been on github since at least 1.4.0, possibly more recent.  It made sense to go through at least the issues that people had brought up here first.  I'll look these things over, but you might want to read the changes file because I see at least a few things we've already fixed.

**Comment by thegushi** [comment]
@stoecker Please go read the CHANGES-202605 file.  I'm looking over your PRs, and at least three of them have already been addressed.

**Comment by thegushi** [comment]
And others already have open issues for tracking.  The existing patches are somewhat helpful, and if they're good, I'll use them, but please stop commenting on a closed issue.

**Comment by thegushi** [comment]
Also, looking at this isn't helpful in evaluating any of this:

<img width="508" height="345" alt="Image" src="https://github.com/user-attachments/assets/3c08a0af-e30c-48fb-ae31-08fe313a479e" />

**Comment by thegushi** [comment]
@stoecker Every patch you're pushing targets the MASTER branch which is not where development happens.  You haven't read about the existing changes, you're just making more work for both of us.

**Comment by thegushi** [comment]
I've unlocked the conversation since you insist on continuing to mention things here.  Because you pushed against master, none of my CI fired (it only fires on the develop branch).  So I'm closing/reopening all PR's just to force that.  That's not a rejection.

**Comment by stoecker** [comment]
> I've unlocked the conversation since you insist on continuing to mention things here.

What would the preferred alternative be? Opening a new ticket for the same topic seemed less appropriate, so I continued in the existing discussion.

> Every patch you're pushing targets the MASTER branch which is not where development happens. You haven't read about the existing changes, you're just making more work for both of us.

I targeted master because it's the default branch on GitHub. I wasn't aware that development happens on develop until after you mentioned it and then I found the little note in the Readme. Had develop been the default branch, I would have based the patches on that branch from the start.

I maintain a large number of RPM packages for openSUSE, so I generally rely on the conventions and defaults provided by each project. Unfortunately I don't always have time to learn the specific workflow of every upstream project before submitting fixes, especially when it differs from the defaults exposed by GitHub.

> Also, looking at this isn't helpful in evaluating any of this:

I referenced the patch names and original ticket numbers because they contain the relevant background and discussion for each change.

**Comment by thegushi** [comment]
So yeah, we should fix the "default branch" thing.  It's already known in my workflow, but for people who are more used to the github ecosystem, I didn't realize that makes it non-obvious.

I've gone through real quick and retitled your existing PR's.  I'm through a few of them now.
